enhancements/installer/azurestack.md

```diff
@@ -0,0 +1,160 @@
+
+Operators will set the `AZURE_ENVIRONMENT_FILEPATH` and mount the endpoints JSON file from the cloud-provider-config configmap. All operators using the Azure SDK will need to do this.
+
+#### Installer Manifests & Cloud Provider Config
+ASH should be able to reuse Azure machine manifests with proper validation.
+
+The [Azure cloud-provider-config](https://github.com/openshift/installer/blob/master/pkg/asset/manifests/azure/cloudproviderconfig.go) will need to be slightly adapted. `LoadBalancerSku` for ASH will be `basic`. `UseInstanceMetadata` should probably be `false`.
+
+The cluster [DNS manifest contains the private DNS zone](https://github.com/openshift/installer/blob/master/pkg/asset/manifests/dns.go#L110-L112). At a minimum this should be removed, but there is an [open question](#open-question) about how this should be handled.
+
+### User Authentication
+
+Users of the installer will authenticate with service principals. The standard Azure workflow works here.
+
+### Lack of Private DNS Zones & Baremetal Networking
+ASH does not provide private DNS zones. The [on-prem, baremetal networking solution](../network/baremetal-networking.md) will be considered to provide private DNS. It remains to be seen whether this can be used with the Azure load balancers.
+
+### Install Config
+New required fields will be added to `installconfig.platform.azure`. At the moment `arm_endpoint` is the only field that will certainly be required. If more than one is required, adding `installconfig.platform.azure.azurestack` should be considered.
+
+Additional fields to be considered would be any required by the private DNS, baremetal networking solution; `subscription_id` is accepted by the azurestack Terraform provider; as are `client_id` and `tenant_id` but these two may be already provided by the service principal.
+
+### Terraform
+Azure Stack has [a stand-alone Terraform provider](https://registry.terraform.io/providers/hashicorp/azurestack/latest/docs).
+
+This provider lacks the ability to create a service principal for the resource group and assign a contributor role, which is required by the Ingress controller.
+
+These actions are achievable through the CLI, if necessary these commands could be run as a Terraform post hook.
+
+The Terraform Azure Active Directory provider does not state support for Azure Stack. It could be tested for AzureStack support, in which case a service principal could be created by that provider and the role assignment would be created with an ARM template executed by the ASH provider.
+
+### Open Questions
+
+1. We need to explore how to pass the JSON endpoints file to the Kubelet:
+    1. Can the endpoints file be simply passed from the installer through ignition? Does this
```

Comment:
Could this be MachineConfigs that the installer adds for master and workers?

Does the bootstrap machine need this as well? I am assuming not since you have been able to run kubelet successfully on the bootstrap machine with the Azure platform.